core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -872,11 +872,12 @@ final class Stream[+F[_], +O] private (private val free: FreeC[Algebra[Nothing,
     }.stream
 
   /**
-    * Tracks any resources acquired during this stream and releases them when the stream completes.
+    * Scopes are typically inserted automatically, at the boundary of a pull (i.e., when a pull
+    * is converted to a stream). This method allows a scope to be explicitly demarcated so that
+    * resources can be freed earlier than when using automatically inserted scopes.
     *
-    * Scopes are sometimes inserted automatically, (e.g., as a result of calling `handleErrorWith`).
-    * This method allows a scope to be explicitly demarcated, so that resources can be freed earlier
-    * than when using automatically inserted scopes.
+    * One use case is scoping the left hand side of an append: `(s1.scope ++ s2)`, which ensures
+    * resources acquired during `s1` are released onces the end of `s1` has been passed.
     */
   def scope: Stream[F, O] = Stream.fromFreeC(Algebra.scope(get))
```

Comment:
Tangential question: what would be a use case for `scope`, now they're back to being inserted automatically on `Pull-to-Stream`?

Reply:
I think my commit just now addresses this question in the updated ScalaDoc.

Reply:
Right, that's roughly what I thought: basically `streamNoScope` and `scope` let you do manual resource management instead of relying on fs2's own mechanism.
I'm ok with this as you never know what people might come up with, and the comment on `streamNoScope` is a sufficient warning :)